src/api/MetadataQuery.js

```diff
@@ -98,32 +123,34 @@ class MetadataQuery extends Base {
      * @return {Object} - flattened metadata entry without the $ fields
      */
     flattenMetadata = (metadata: MetadataQueryResponseEntryMetadata): FlattenedMetadataQueryResponseEntryMetadata => {
-        let flattenedMetadata = {};
-
-        Object.keys(metadata).forEach(scope => {
-            Object.keys(metadata[scope]).forEach(templateKey => {
-                const nonconformingInstance = metadata[scope][templateKey];
-                const data = omit(nonconformingInstance, [
-                    '$id',
-                    '$parent',
-                    '$type',
-                    '$typeScope',
-                    '$typeVersion',
-                    '$version',
-                ]);
-
-                flattenedMetadata = {
-                    data,
-                    id: nonconformingInstance.$id,
-                    metadataTemplate: {
-                        type: 'metadata-template',
-                        templateKey,
-                    },
+        const templateFields = getProp(this.templateInfo, 'fields', []);
+        const instance = getProp(metadata, `${this.templateScope}.${this.templateKey}`);
+
+        if (!instance) {
+            return {};
+        }
+
+        const fields = Object.keys(instance)
+            .filter(key => !key.startsWith('$'))
+            .map(key => {
+                const type = getProp(find(templateFields, ['key', key]), 'type'); // get data type
+                const field: FlattenedMetadataQueryResponseEntryMetadataField = {
+                    name: key,
+                    value: instance[key],
+                    type,
                 };
+
+                if ([FIELD_TYPE_ENUM, FIELD_TYPE_MULTISELECT].includes(type)) {
```

Comment:
Could this instead be a constant defined above like `SELECT_TYPES` and referenced here? `SELECT_TYPES.includes(type)`